dashboard/legacy/middleware/helpers/projects.rb

```diff
@@ -448,6 +451,22 @@ def self.in_restricted_share_mode(channel_id, project_type)
 
   private
 
+  # TODO: post-firebase-cleanup, remove this once we switch 100% to datablock storage
+  def set_use_datablock_storage(project_id, project_type)
+    if DatablockStorageController::SUPPORTED_PROJECT_TYPES.include? project_type
+      # While we transition, a fraction of new projects will be set at creation
+      # to use datablock storage. As we gain confidence, we can increase this
+      # DCDO flag to 1.0. At that point, we're ready to migrate all the old projects.
+      #
+      # Once 100% of the old projects are migrated, we're ready to remove code
+      # marked with: TODO: post-firebase-cleanup
+      ProjectUseDatablockStorage.find_or_create_by(project_id: project_id) do |storage|
+        fraction = DCDO.get('fraction_of_new_projects_use_datablock_storage', 0.0)
+        storage.use_datablock_storage = rand < fraction
+      end
+    end
+  end
+
```

Comment:
After merging this PR, whenever we create a new Applab project, we roll the dice and check against `DCDO.get('fraction_of_new_projects_use_datablock_storage')` to decide if the new project will be backed by datablock storage or firebase. This allows us to do a gradual rollout and see how load etc works out at production scale.